### PR TITLE
btl/openib: add XRC support with OFED 3.12+

### PR DESCRIPTION
**Pull request by ggouaillardet**
based on an original patch contributed by Bull.

### CODE REVIEW

**Comment by vasilyMellanox** [comment]
Looks good to me

File: opal/mca/btl/openib/btl_openib.h [hunk 1]
```diff
@@ -149,6 +150,12 @@ typedef struct mca_btl_openib_srq_manager_t {
 } mca_btl_openib_srq_manager_t;
 #endif
 
+typedef enum {
+    MCA_BTL_IB_XRC_API_NONE,
+    MCA_BTL_IB_XRC_API_BETA,
+    MCA_BTL_IB_XRC_API_OFED_3_12
+} mca_btl_openib_xrc_api_t;
```
Comment (jsquyres) [inline opal/mca/btl/openib/btl_openib.h]: Is it possible to use names that are not dependent on specific versions?

E.g., what does "beta" refer to?  Is that a beta of a specific OFED (or MOFED) version?

Is it possible to have names that reflect specific API characteristics?

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/161/
Test PASSed.

**Comment by ggouaillardet** [comment]
@jsquyres
i made updated, rebased and pushed a new version of this PR
basically, i revamped the runtime check, and replace the
OPAL_HAVE_CONNECTIB_XRC macro with OPAL_HAVE_XRCD which makes more sense

**Comment by jsquyres** [comment]
@ggouaillardet Looks good -- thanks!

**Comment by jsquyres** [comment]
We discussed this on the call today, and it looks like everyone is happy -- so I'll just merge this in.

**Pull request merged**